static/app/components/profiling/Flamegraph.tsx

```diff
@@ -0,0 +1,134 @@
+import {useFlamegraphPreferences} from 'sentry/utils/profiling/flamegraph/useFlamegraphPreferences';
+import {ProfileGroup} from 'sentry/utils/profiling/profile/importProfile';
+
+interface FlamegraphProps {
+  profiles: ProfileGroup;
+}
+
+function Flamegraph(props: FlamegraphProps): React.ReactElement {
+  const canvasPoolManager = useMemo(() => new CanvasPoolManager(), []);
+  const [{sorting, view, colorCoding}, dispatch] = useFlamegraphPreferences();
+
+  const [flamegraph, setFlamegraph] = useState(
+    new FlamegraphModel(
+      props.profiles.profiles[0],
+      0,
+      view === 'bottom up',
+      sorting === 'left heavy'
+    )
+  );
+
+  const onImport = useCallback(
+    profile => {
+      setFlamegraph(
+        new FlamegraphModel(
+          profile.profiles[0],
+          0,
+          view === 'bottom up',
+          sorting === 'left heavy'
+        )
+      );
+    },
+    [view === 'bottom up', sorting === 'left heavy']
+  );
+
+  const onProfileIndexChange = useCallback(
+    index => {
+      setFlamegraph(
+        new FlamegraphModel(
+          props.profiles.profiles[index],
+          index,
+          view === 'bottom up',
+          sorting === 'left heavy'
+        )
+      );
+    },
+    [view === 'bottom up', sorting === 'left heavy']
+  );
+
+  useEffect(() => {
+    setFlamegraph(
+      new FlamegraphModel(
+        props.profiles.profiles[0],
+        0,
+        view === 'bottom up',
+        sorting === 'left heavy'
+      )
+    );
+  }, [view === 'bottom up', sorting === 'left heavy']);
+
+  return (
+    <FlamegraphThemeProvider>
+      <div
+        style={{
+          display: 'flex',
+          flexDirection: 'column',
+          height: `100vh`,
+          overflow: 'hidden',
+          overscrollBehavior: 'contain',
+        }}
+      >
+        <FlamegraphToolbar>
+          <FlamegraphViewSelectMenu
+            view={view === 'bottom up' ? 'bottom up' : 'top down'}
+            sorting={sorting === 'left heavy' ? 'left heavy' : 'call order'}
+            onSortingChange={s => {
+              dispatch({type: 'set sorting', value: s});
+            }}
+            onViewChange={v => {
+              dispatch({type: 'set view', value: v});
+            }}
+          />
+          <ThreadMenuSelector
+            profileGroup={props.profiles}
+            activeProfileIndex={flamegraph.profileIndex}
+            onProfileIndexChange={onProfileIndexChange}
+          />
+          <FlamegraphOptionsMenu
+            colorCoding={colorCoding}
+            onColorCodingChange={c => dispatch({type: 'set color coding', value: c})}
+            canvasPoolManager={canvasPoolManager}
+          />
+        </FlamegraphToolbar>
+        <div style={{height: 100, position: 'relative'}}>
+          <FlamegraphZoomViewMinimap
+            flamegraph={flamegraph}
+            highlightRecursion={colorCoding === 'by recursion'}
```

Comment:
Thoughts on removing this `highlightRecursion` prop?
1. it's not being used internally by `FlamegraphZoomViewMinimap` or `FlamegraphZoomView`
2. the `colorCoding` prop already contains the same information (despite not being used at the moment)

Reply:
yeah, lets remove it 100%